src/runtime/server/api/query.ts

```diff
@@ -1,11 +1,26 @@
 import { createError, defineEventHandler } from 'h3'
+import { join } from 'pathe'
 import { serverQueryContent } from '../storage'
 import { getContentQuery } from '../../utils/query'
 
 export default defineEventHandler(async (event) => {
   const query = getContentQuery(event)
   const contents = await serverQueryContent(event, query).find()
 
+  if (query.first) {
+    const path = contents?._path || query.where.find(w => w._path)?._path
+    if (path) {
+      const _dir = await serverQueryContent(event).where({ _path: join(path, '_dir') }).without('_').findOne()
+      if (!Array.isArray(_dir)) {
```

Comment:
Why checking if it is not an array? In which case `findOne()` returns an Array?

Reply:
It returns an empty array only when there is no matched document.
We will fix this behavior in another PR